code/modules/mob/living/carbon/brain/brain_item.dm

```diff
@@ -76,12 +73,12 @@
 
 	..()
 
-/obj/item/organ/brain/replaced(var/mob/living/target)
+/obj/item/organ/internal/brain/insert(var/mob/living/target)
 
 	var/brain_already_exists = 0
 	if(istype(target,/mob/living/carbon/human)) // No more IPC multibrain shenanigans
-		var/mob/living/carbon/human/H = target
-		if(organ_tag in H.internal_organs_by_name)
+		//var/mob/living/carbon/human/H = target
+		if(target.get_int_organ(/obj/item/organ/internal/brain))

```

Comment:
Uhh IPC brains, as-is, don't exist under the organ/internal/brain path, so this would just cause the problems again.
Could always try moving the MMI holder organ to a subpath of brain, and override a few key procs so things go swimmingly, though.